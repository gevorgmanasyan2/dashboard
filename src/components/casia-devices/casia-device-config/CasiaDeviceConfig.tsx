import { RightOutlined } from '@ant-design/icons'
import React, { FC, Fragment, useState } from 'react'
import casia1Img from '../../../assets/images/casia1.png'
import casia2Img from '../../../assets/images/casia2.png'
import './casiaDeviceConfig.scss'
import ActionButtons from '../../action-buttons/ActionButtons'
import { Link } from 'react-router-dom'
import Title from '../../title/Title'

const casiaData = [
  {
    id: 1,
    title: 'Casia 360',
    serial: 'H234 12348970123',
    img: casia1Img,
  },
  {
    id: 2,
    title: 'Casia Long Range',
    serial: 'H250 12898970123',
    img: casia2Img,
  },
]

const CasiaDeviceConfig: FC = () => {
  const [isDisabled, setIsDisabled] = useState<boolean>(true)
  return (
    <article className="config__container">
      <section className="config__container__title">
        <Title titleText="Un-configured Casia Devices" level={2} />
        <p onClick={() => setIsDisabled(false)}>Select the device you want to configure:</p>
      </section>
      <section className="config__container__casia">
        {casiaData.map((casia) => (
          <Fragment key={casia.id}>
            <section
              className="config__container__casia__devices"
              onClick={() => setIsDisabled(false)}
            >
              <div className="config__container__casia__devices__element">
                <p className="config__container__casia__devices__element--title">{casia.title}</p>
                <p className="config__container__casia__devices__element--serial">
                  Serial #: {casia.serial}
                </p>
              </div>
              <div className="config__container__casia__devices__image">
                <img src={casia.img} alt="" />
              </div>
            </section>
          </Fragment>
        ))}
      </section>
      <section className="config__container__buttons">
        <Link to="/">
          <ActionButtons
            className="config__container__buttons--cancel"
            buttonText="Back"
          ></ActionButtons>
        </Link>
        <Link to="/casiadevice-form">
          <ActionButtons
            className="config__container__buttons--next"
            icon={<RightOutlined className="right__arrow" />}
            buttonType={isDisabled}
            buttonText="Next"
          ></ActionButtons>
        </Link>
      </section>
    </article>
  )
}

export default CasiaDeviceConfig
