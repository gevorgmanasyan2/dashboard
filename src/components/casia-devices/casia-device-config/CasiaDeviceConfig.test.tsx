/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import CasiaDeviceConfig from './CasiaDeviceConfig'

it('should render CasiaDeviceConfig component', () => {
  expect(shallow(<CasiaDeviceConfig />)).toMatchSnapshot()
})

// test('should call setItemDisabled method', () => {
//   const wrapper = shallow(<CasiaDeviceConfig />)
//   const params = {isDisabled:true}
//   const instance = wrapper.instance()  CasiaDeviceConfig
//   const value = true
//   instance.setItemDisabled(params,false)

//   expect(wrapper.instance().setIsDisabled.has(1)).toBeTruthy()


//   wrapper.instance().setItemDisabled(params, false)


//   expect(wrapper.instance().setIsDisabled.has(1)).toBeFalsy()
// })