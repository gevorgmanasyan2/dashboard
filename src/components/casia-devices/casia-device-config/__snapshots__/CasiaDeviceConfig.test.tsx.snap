// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CasiaDeviceConfig component 1`] = `
<article
  className="config__container"
>
  <section
    className="config__container__title"
  >
    <Title
      level={2}
      titleText="Un-configured Casia Devices"
    />
    <p
      onClick={[Function]}
    >
      Select the device you want to configure:
    </p>
  </section>
  <section
    className="config__container__casia"
  >
    <section
      className="config__container__casia__devices"
      onClick={[Function]}
    >
      <div
        className="config__container__casia__devices__element"
      >
        <p
          className="config__container__casia__devices__element--title"
        >
          Casia 360
        </p>
        <p
          className="config__container__casia__devices__element--serial"
        >
          Serial #: 
          H234 12348970123
        </p>
      </div>
      <div
        className="config__container__casia__devices__image"
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
    </section>
    <section
      className="config__container__casia__devices"
      onClick={[Function]}
    >
      <div
        className="config__container__casia__devices__element"
      >
        <p
          className="config__container__casia__devices__element--title"
        >
          Casia Long Range
        </p>
        <p
          className="config__container__casia__devices__element--serial"
        >
          Serial #: 
          H250 12898970123
        </p>
      </div>
      <div
        className="config__container__casia__devices__image"
      >
        <img
          alt=""
          src="test-file-stub"
        />
      </div>
    </section>
  </section>
  <section
    className="config__container__buttons"
  >
    <Link
      to="/"
    >
      <ActionButtons
        buttonText="Back"
        className="config__container__buttons--cancel"
      />
    </Link>
    <Link
      to="/casiadevice-form"
    >
      <ActionButtons
        buttonText="Next"
        buttonType={true}
        className="config__container__buttons--next"
        icon={
          <ForwardRef(RightOutlined)
            className="right__arrow"
          />
        }
      />
    </Link>
  </section>
</article>
`;
